src/utils.rs

```diff
@@ -1,3 +1,42 @@
+use core::fmt;
+
+use asn1_rs::{BmpString, Choice, PrintableString, TeletexString, UniversalString, Utf8String};
+
+/// The DirectoryString type is defined as a choice of PrintableString, TeletexString,
+/// BMPString, UTF8String, and UniversalString.
+///
+/// <pre>
+/// RFC 5280, 4.1.2.4.  Issuer
+///    DirectoryString ::= CHOICE {
+///          teletexString           TeletexString (SIZE (1..MAX)),
+///          printableString         PrintableString (SIZE (1..MAX)),
+///          universalString         UniversalString (SIZE (1..MAX)),
+///          utf8String              UTF8String (SIZE (1..MAX)),
+///          bmpString               BMPString (SIZE (1..MAX))
+///    }
+/// </pre>
+#[derive(Debug, PartialEq, Eq, Choice)]
+#[asn1(parse = "DER", encode = "")]
+pub enum DirectoryString<'a> {
```

Comment:
It feels like a bit of an odd choice for `DirectoryString` to end up in a `utils.rs`. I'm generally skeptical of "util" modules since they can sort of end up as a catch-all junkyard over time.

In this case it feels like both `format_serial` and the `DirectoryString` type would be more at home in `x509.rs`. WDYT?

Reply:
I agree on the "catch-all junkyard" feeling :)
I'll move these to `x509.rs`. Also, `format_serial` could be moved to an implementation of `LowerHex`.

Reply:
> This makes me wonder if we should add a Serial object to encapsulate the raw bytes

That sounds reasonable to me!